File: backend/src/main/java/ResumeBuilder/backend/Models/DatabaseModels/ExperienceModel.java
package ResumeBuilder.backend.Models.DatabaseModels;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import ResumeBuilder.backend.Interfaces.IWriter;
import ResumeBuilder.backend.Utilities.DateFormatUtility;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "experience")
public class ExperienceModel implements IWriter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public int id;

    @Column
    public String title;

    @Column
    public LocalDate startDate;

    @Column
    public LocalDate endDate;

    @Column
    public String company;

    @Column
    public String location;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "experience_id")
    public List<ExperienceBulletPointModel> description;

    public Object get(String key) {
        if (key.equals("title")) {
            return title;
        } else if (key.equals("startDate")) {
            return DateFormatUtility.toString(startDate);
        } else if (key.equals("endDate")) {
            return DateFormatUtility.toString(endDate);
        } else if (key.equals("company")) {
            return company;
        } else if (key.equals("location")) {
            return location;
        } else if (key.equals("description")) {
            return description;
        }
        return null;
    }
}

File: frontend/models/ExperienceModel.js
import ExperienceBulletPointModel from "./ExperienceBulletPointModel";
import { v4 as uuid } from "uuid";

export default class ExperienceModel {
  id = uuid();
  title = "";
  startDate;
  endDate;
  company = "";
  location = "";
  description = new Array();

  constructor(data) {
    if (data != null) {
      this.id = data.id;
      this.title = data.title;
      this.startDate = data.startDate;
      this.endDate = data.endDate;
      this.company = data.company;
      this.location = data.location;
      this.description = data.description.map((item) => {
        return new ExperienceBulletPointModel(item);
      });
    }
  }
}
